File: server-pedidos-java/src/main/java/br/edu/utfpr/serverpedidosjava/util/validator/TelefoneValidator.java
package br.edu.utfpr.serverpedidosjava.util.validator;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

public class TelefoneValidator implements ConstraintValidator<Telefone, String> {

    @Override
    public boolean isValid(String value, ConstraintValidatorContext context) {
        return value == null || value.isEmpty() || value.matches("^(\\(\\d{2}\\)|\\d{2})\\s?9?\\d{4}\\-?\\d{4}$");
    }

}

File: server-pedidos/src/main/kotlin/br/edu/utfpr/serverpedidos/util/validator/TelefoneValidator.kt
package br.edu.utfpr.serverpedidos.util.validator

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext

class TelefoneValidator : ConstraintValidator<Telefone, String> {
    override fun isValid(value: String?, context: ConstraintValidatorContext?): Boolean {
        return value.isNullOrEmpty() ||
                value.matches(Regex("^(\\(\\d{2}\\)|\\d{2})\\s?9?\\d{4}\\-?\\d{4}\$"))
    }
}
